tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -666,6 +666,45 @@ final class TestingServer(
     }
   }
 
+  def assertTypeDefinition(
+      filename: String,
+      query: String,
+      expectedPos: Location
+  ): Future[Unit] = {
+    def locationToString(loc: l.Location): String = {
+      val (uri, start, end) =
+        (loc.getUri, loc.getRange.getStart, loc.getRange.getEnd)
+      s"|$uri/\\${start.getLine}:${start.getCharacter}/\\${end.getLine}:${end.getCharacter}|"
+    }
+
+    for {
+      typeDefinitions <- typeDefinition(filename, query).map(
+        _.map(locationToString)
+      )
+    } yield {
+      DiffAssertions.assertNoDiff(
+        typeDefinitions.headOption.getOrElse(""),
+        locationToString(expectedPos)
+      )
+    }
+  }
+
+  def typeDefinition(
+      filename: String,
+      query: String
+  ): Future[List[l.Location]] = {
+
+    for {
+      (_, params) <- offsetParams(filename, query, workspace)
+      typeDefinitions <- server
```

Comment:
I would change it to `definitions` really, because it can be mixed up with the method name.